warehouse/accounts/views.py

```diff
@@ -449,6 +456,47 @@ def webauthn_authentication_validate(request):
     return {"fail": {"errors": errors}}
 
 
+def _check_remember_device_token(request, user_id) -> bool:
+    """
+    Returns true if the given remember device cookie is valid for the given user.
+    """
+    remember_device_token = request.cookies.get(REMEMBER_DEVICE_COOKIE)
+    if not remember_device_token:
+        return False
+    token_service = request.find_service(ITokenService, name="remember_device")
+    try:
+        data = token_service.loads(remember_device_token)
+        user_id_token = data.get("user_id")
+        return user_id_token == str(user_id)
+    except TokenException:
+        return False
+
+
+def _remember_device(request, response, userid, two_factor_method) -> None:
+    """
+    Generates and sets a cookie for remembering this device.
+    """
+    remember_device_data = {"user_id": str(userid)}
+    token_service = request.find_service(ITokenService, name="remember_device")
+    token = token_service.dumps(remember_device_data)
+    response.set_cookie(
+        REMEMBER_DEVICE_COOKIE,
+        token,
+        max_age=REMEMBER_DEVICE_SECONDS,
+        httponly=True,
+        secure=request.scheme == "https",
+        samesite=b"lax",
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏can you help explain why we need to allow for lax (the default), vs strict? I think you're following the pattern in sessions.py, but I think we might be able to tighten this one up a bit more.

Reply:
Your right that I was following the pattern from session.py. For this cookie, we don't need lax, but I also don't think we would benefit from strict. Since this cookie is restricted to `account/login`, if we set it to lax and the user navigates to `https://pypi.org/account/login` from another page, this cookie would be sent, but it wouldn't be used the server since we only use it on POST requests. If we set it to strict, the only difference is that it wouldn't be sent when the user first navigates to `https://pypi.org/account/login` from another page. I'll change it to strict because in general I think one should use strict unless one has a need for lax.